src/adera/Shaders/Phong.cpp

```diff
@@ -13,7 +13,7 @@ void Phong::draw(ActiveEnt e,
     PhongShaderInstance& shaderInstance = scene.reg_get<PhongShaderInstance>(e);
     Phong& shader = *shaderInstance.m_shaderProgram;
 
-    Magnum::Matrix4& entRelative = camera.m_inverse * transform.m_transformWorld;
+    Magnum::Matrix4 entRelative = camera.m_inverse * transform.m_transformWorld;
```

Comment:
so if you were to use const& here to bind the entRelative variable to the temporary, and cause lifetime extension, then that technically allows for a situation where "camera.m_inverse * transform.m_transformWorld" returns a reference to a Matrix4 that exists elsewhere, or a situation where "camera.m_inverse * transform.m_transformWorld" returns a polymorphic object that has base class Matrix4.

Not saying you should do that, just pointing out the nifty detail of the language.

You can additionally do && instead of const&, and get lifetime extension that way, and then std::move() it later.

But my understanding is that matrix4 is a small POD type so doing that wouldn't do any good.

Reply:
It's fine by-value like this; I have zero idea why it compiles and runs with MSVC as it was. The Windows build passed (and it ran on my machine); I wouldn't have noticed anything if the linux build hadn't failed. I need to always remember to re-enable `permissive-` mode.